src/BuildScriptGeneratorCli/Commands/TelemetryCommand.cs

```diff
@@ -0,0 +1,81 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Logs an event in Oryx telemetry.")]
+    internal class TelemetryCommand : CommandBase
+    {
+        public const string Name = "telemetry";
+
+        [Option(
+            OptionTemplates.EventName,
+            CommandOptionType.SingleValue,
+            Description = "The name of the event to log for telemetry.")]
+        public string EventName { get; set; }
+
+        [Option(
+            OptionTemplates.ProcessingTime,
+            CommandOptionType.SingleValue,
+            Description = "The processing time of the event, in milliseconds")]
+        public double ProcessingTime { get; set; }
+
+        [Option(
+            OptionTemplates.Property,
+            CommandOptionType.MultipleValue,
+            Description = "Additional information that should be logged with the event, in the form --property KEY=VALUE.")]
```

Comment:
So just curious: in practice this would be used as `oryx telemetry XYZ --property A=1 --property B=2 --property C=3`?

Reply:
That's correct -- it gives us a little more flexibility when logging events to provide `customDimensions` without needing to represent them as their own arguments for the command.

Reply:
Thanks!